src/app/public/modules/text-editor/services/text-editor-adapter.service.ts

```diff
@@ -0,0 +1,532 @@
+          return false;
+        }
+        const tempRange = document.createRange();
+        tempRange.selectNodeContents(element);
+        return (selectedRange.compareBoundaryPoints(Range.START_TO_START, tempRange) !== -1
+          && selectedRange.compareBoundaryPoints(Range.START_TO_END, tempRange) !== 1) ||
+          (selectedRange.compareBoundaryPoints(Range.END_TO_START, tempRange) !== -1
+          && selectedRange.compareBoundaryPoints(Range.END_TO_END, tempRange) !== 1);
+      }
+
+      // Normal non-IE
+      return !!element && !!element.href && selectedRange.intersectsNode(element);
+    });
+  }
+
+  private getDocumentEl(id: string): Document {
+    /* istanbul ignore next */
+    if (!(id in this.editors)) {
+      return undefined;
+    }
+
+    const contentWindowEl = this.getContentWindowEl(id);
+    if (contentWindowEl) {
+      return contentWindowEl.document;
+    }
+
+    /* istanbul ignore next */
+    return this.editors[id].iframeElementRef.contentDocument;
+  }
+
+  private getFontSize(id: string): string {
+    /* istanbul ignore next */
+    let fontSize = STYLE_STATE_DEFAULTS.fontSize.toString();
+    const selection = this.getCurrentSelection(id);
+    if (
+      selection &&
+      selection.anchorNode &&
+      selection.anchorNode.parentElement
+    ) {
+      let element = selection.anchorNode;
+      if (element.nodeType !== 1) {
+        element = element.parentElement;
+      }
+      const computedStyle = window.getComputedStyle(element as Element);
+      /* istanbul ignore else */
+      if (computedStyle) {
+        fontSize = computedStyle.getPropertyValue('font-size');
+      }
+    }
+    return fontSize;
+  }
+
+  private createObservers(id: string, element: HTMLIFrameElement): EditorSetting {
+    /* istanbul ignore next */
+    const documentEl = element.contentWindow ?
+      element.contentWindow.document :
+      element.contentDocument;
+
+    // Firefox bug where we need to open/close to cancel load so it doesn't overwrite attrs
+    documentEl.open();
+    documentEl.close();
+
+    const selectionObservable = new Subject();
+    const selectionListener = () => selectionObservable.next();
+    const clickObservable = new Subject();
+    const clickListener = () => clickObservable.next();
+    const pasteListener = this.getPasteOverride(id);
+
+    documentEl.addEventListener('selectionchange', selectionListener);
+    documentEl.addEventListener('input', selectionListener);
+    documentEl.addEventListener('mousedown', clickListener);
+    documentEl.body.addEventListener('paste', pasteListener);
+    return {
+      iframeElementRef: element,
+      selectionListener: selectionListener,
+      clickListener: clickListener,
+      pasteListener: pasteListener,
+      selectionChangeObservable: selectionObservable,
+      clickObservable: clickObservable,
+      commandChangeObservable: new Subject()
+    };
+  }
+
+  public static removeObservers(setting: EditorSetting): void {
```

Comment:
I'm still not sure that I love this being static - but I won't harp on that. However, it is out of order with this public method being after other private methods.

Reply:
Talked offline - reverted to non-static methods.